File: DONGUK/프로그래머스_WEEK 5/폰켓몬.py
def solution(nums):
    answer = 0
    d = {}
    
    for num in nums:
        if num not in d:
            d[num] = 1
        else:
            d[num] += 1
    
    if len(d) <= len(nums) // 2:
        answer = len(d)
    
    else:
        answer = len(nums) // 2 
        
    return answer

File: JUNHEE/프로그래머스_week5/폰켓몬.cpp
#include <vector>
using namespace std;

int solution(vector<int> nums)
{
    int length = nums.size();
    vector<int> ans;
    int idx = 0;
    for(int i = 0; i < length; i++){
        bool flag = true;
        for(int j = 0; j < ans.size();j++){
            if(ans[j] == nums[i]){
                flag = false;
                break;
            }
        }
        if(flag){
            ans.push_back(nums[i]);
        }
    }
    if(ans.size() > length / 2) return length/2;
    else return ans.size();
}
